src/commands/deploy.rs

```diff
@@ -299,12 +301,15 @@ impl DeployCommand {
             .id)
     }
 
-    async fn get_channel_id(&self, hippo_client: &Client, name: String) -> Result<Uuid> {
+    async fn get_channel_id(&self, hippo_client: &Client, app_id: Uuid) -> Result<Uuid> {
         let channels_vm = Client::list_channels(hippo_client).await?;
-        let channel = channels_vm.items.iter().find(|&x| x.name == name.clone());
+        let channel = channels_vm
+            .items
+            .iter()
+            .find(|&x| x.app_id == app_id.clone());
```

Comment:
Can you check for both the channel name as well as the app id?

```
async fn get_channel_id(&self, hippo_client: &Client, app_id: Uuid, name: String) -> Result<Uuid> {
...
let channel = channels_vm
             .items
             .iter()
             .find(|&x| x.app_id == app_id.clone() && x.name == name.clone());
```

Reply:
Since channel name is always `spin_deploy` if pushed with `spin deploy`, i dont see when we would get multiples here but it makes sense to prevent an issue here in the future where we may find multiple if channel name varies

Reply:
Users could add more channels in the UI as well. Those channel names don't have to match "spin-deploy"